### ISSUE TITLE
Dashboard: My Stories; A11y improvements

### ISSUE THREAD

**Issue by carloskelly13** [GoogleForCreators/web-stories-wp#4628]
## Feature Description

1. List View: The headers should announce as buttons to show users using screen readers that these are to sort the table.
2. Delete Dialog: Buttons should have context, like "Confirm you wish to delete X story?"

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

**Comment by csossi** [comment]
The ChromeVox audio for this section doesn't match what is specified in #4642

Currently has:
"Cancel deleting story quote _story name_ quote"
and
"Confirming deleting story quote _story name_ quote"

**Comment by BrittanyIRL** [comment]
@csossi  - This is my fault for not updating this to match discussion in the PR. 
https://github.com/google/web-stories-wp/pull/4642#discussion_r493714039 

Pascal pointed out that full sentences sounded a little out of place,  so we updated it to be:
"Cancel deleting story quote story name quote"
and
"Confirming deleting story quote story name quote"

So what you heard is actually correct.

**Comment by csossi** [comment]
Verified in QA
